lib/dialects/oracledb/index.js

```diff
@@ -165,6 +180,58 @@ class Client_Oracledb extends Client_Oracle {
     return connection.release();
   }
 
+  // In Oracle, we need to check the version to dynamically determine
+  // certain limits.
+  async checkVersion(connection) {
+    // Already determined version before?
+    if (this.version) {
+      return this.version;
+    }
+
+    return new Promise((resolver, rejecter) => {
+      connection.execute(`select BANNER from V$VERSION`, (err, response) => {
+        if (err) {
+          return rejecter(err);
+        }
+
+        const versionString = response.rows[0][0];
+        const detectedVersion = this._parseVersion(versionString);
+
+        if (!detectedVersion) {
+          return rejecter(
+            // When original version is set to null, user-provided version was invalid and we fell-back to auto-detect
+            // Otherwise, we couldn't auto-detect.
+            new Error(
+              this.version === null
+                ? 'Invalid Oracledb version number format passed to knex. Unable to successfully auto-detect as fallback.'
+                : 'Unable to detect Oracledb version number automatically. Please specify the version in knex configuration.'
+            )
+          );
+        }
+
+        resolver(detectedVersion);
+      });
+    });
+  }
+
+  _parseVersion(versionString) {
+    try {
+      const version = /^\d/.test(versionString)
+        ? versionString
+        : /^Oracle Database .* ((?:\d+\.){4}\d+).*$/.exec(versionString)[1];
+      // We only care about first two version components at most
+      const versionParts = version.split('.').slice(0, 2);
+      // Strip off any character suffixes in version number (ex. 12c => 12, 12.2c => 12.2)
+      versionParts.forEach((versionPart, idx) => {
+        versionParts[idx] = versionPart.replace(/\D/, '');
+      });
+      return versionParts.join('.');
+    } catch {
```

Comment:
missing error variable:

`} catch(err) {
`

Reply:
@OlivierCavadenti This has been fixed.